### PR TITLE
Nrfx 4851 replace checking gpio instance number

### PR DESCRIPTION
**Pull request by magp-nordic**
~~Based on https://github.com/NordicSemiconductor/npmx-zephyr/pull/117~~ Rebased on develop
Please, review only the last commit.

### CODE REVIEW

**Review by masz-nordic — DISMISSED** [review]

File: drivers/npmx/shell/buck.c [hunk 1]
```diff
@@ -76,6 +76,26 @@ static int cmd_buck_active_discharge_get(const struct shell *shell, size_t argc,
 	return 0;
 }
 
+static npmx_buck_gpio_t buck_gpio_index_convert(int32_t gpio_idx)
```
Comment (wilu-nordic) [inline drivers/npmx/shell/buck.c:79]: Please rename it to `gpio_index`. Try not to use any short names when it is possible.

**Review by wilu-nordic — DISMISSED** [review]

**Review by masz-nordic — APPROVED** [review]

**Review by wilu-nordic — APPROVED** [review]

**Comment by nordic-segl** [comment]
Can't run test, precommit is failing. Most likely, rebase will fix issues.
https://jenkins-ncs.nordicsemi.no/blue/organizations/jenkins/HCS%2FNPMX%2Fnpmx_zephyr_integration_downstream/detail/PR-202/8/pipeline/200

**Review by wilu-nordic — DISMISSED** [review]

**Comment by nordic-segl** [comment]
I see invalid error messages
uart:~$ npmx buck gpio on_off index set 0 256
Error: GPIO used as interrupt.
uart:~$ npmx buck gpio on_off index set 0 257
Error: GPIO used as POF.
uart:~$ npmx buck gpio on_off index set 0 258
Error: wrong GPIO index.

**Comment by magp-nordic** [comment]
> I see invalid error messages uart:~$ npmx buck gpio on_off index set 0 256 Error: GPIO used as interrupt. uart:~$ npmx buck gpio on_off index set 0 257 Error: GPIO used as POF. uart:~$ npmx buck gpio on_off index set 0 258 Error: wrong GPIO index.

Fix is in the last commit.

**Review by masz-nordic — APPROVED** [review]

**Comment by nordic-segl** [comment]
Waiting for results from nightly test scope:
https://jenkins-ncs.nordicsemi.no/job/HCS/job/NPMX/job/npmx_zephyr_develop_samples/job/main/325/

**Review by nordic-segl — APPROVED** [review]

**Pull request merged**